trunk/src/app/srs_app_rtc_source.cpp

```diff
@@ -2853,25 +3065,44 @@ SrsRtcVideoRecvTrack::~SrsRtcVideoRecvTrack()
 {
 }
 
-void SrsRtcVideoRecvTrack::on_before_decode_payload(SrsRtpPacket* pkt, SrsBuffer* buf, ISrsRtpPayloader** ppayload, SrsRtspPacketPayloadType* ppt)
+void SrsRtcVideoRecvTrack::on_before_decode_payload(SrsRtpPacket* pkt, SrsBuffer* buf, ISrsRtpPayloader** ppayload, SrsRtpPacketPayloadType* ppt)
 {
     // No payload, ignore.
     if (buf->empty()) {
         return;
     }
 
-    uint8_t v = (uint8_t)(buf->head()[0] & kNalTypeMask);
-    pkt->nalu_type = v;
+    SrsVideoCodecId video_codec = srs_video_codec_str2id(track_desc_->media_->name_);
+    if (video_codec == SrsVideoCodecIdAVC) {
+        uint8_t v = SrsAvcNaluTypeParse(buf->head()[0]);
+        pkt->nalu_type = v;
 
-    if (v == kStapA) {
-        *ppayload = new SrsRtpSTAPPayload();
-        *ppt = SrsRtspPacketPayloadTypeSTAP;
-    } else if (v == kFuA) {
-        *ppayload = new SrsRtpFUAPayload2();
-        *ppt = SrsRtspPacketPayloadTypeFUA2;
-    } else {
-        *ppayload = new SrsRtpRawPayload();
-        *ppt = SrsRtspPacketPayloadTypeRaw;
+        if (v == kStapA) {
+            *ppayload = new SrsRtpSTAPPayload();
+            *ppt = SrsRtpPacketPayloadTypeSTAP;
+        } else if (v == kFuA) {
+            *ppayload = new SrsRtpFUAPayload2();
+            *ppt = SrsRtpPacketPayloadTypeFUA2;
+        } else {
+            *ppayload = new SrsRtpRawPayload();
+            *ppt = SrsRtpPacketPayloadTypeRaw;
+        }
+#ifdef SRS_H265
+    } else if (video_codec == SrsVideoCodecIdHEVC) {
+        uint8_t v = SrsHevcNaluTypeParse(buf->head()[0]);
+        pkt->nalu_type = v;
+
+        if (v == kStapHevc) {
+            *ppayload = new SrsRtpSTAPPayloadHevc();
+            *ppt = SrsRtpPacketPayloadTypeSTAPHevc;
+        } else if (v == kFuHevc) {
+            *ppayload = new SrsRtpFUAPayloadHevc2();
+            *ppt = SrsRtpPacketPayloadTypeFUAHevc2;
+        } else {
+            *ppayload = new SrsRtpRawPayload();
+            *ppt = SrsRtpPacketPayloadTypeRaw;
+        }
+#endif
```

Comment:
If not AVC or HEVC, the payload and pt is undefined. You can:
1. Use assert to make sure no problem.
2. Use SrsRtpPacketPayloadTypeRaw by default.